File: src/worker/core/season/newSchedule.java
import static common.Index.WEBSITE_ROOT;
import common.Types.Conditions;
import util.Index.G;
import util.Index.Helpers;
import util.Index.LogEvent;
import static newScheduleGood.NewScheduleGood.newScheduleGood;

import java.util.List;

public class NewSchedule {

    public static int[][] createNewSchedule(List<Team> teams, Conditions conditions) {
        int[][] tids;
        String warning;

        // Fetch tids and warning from newScheduleGood function
        int[][] result = newScheduleGood(teams);
        tids = result[0];
        warning = result[1];

        // Add trade deadline
        int tradeDeadline = G.get("tradeDeadline");
        if (tradeDeadline < 1) {
            int index = Math.round(Helpers.bound(tradeDeadline, 0, 1) * tids.length);
            tids = insertAtIndex(tids, new int[]{-3, -3}, index);
        }

        // Add an All-Star Game
        Integer allStarGame = G.get("allStarGame");
        if (allStarGame != null && allStarGame >= 0) {
            int index = Math.round(Helpers.bound(allStarGame, 0, 1) * tids.length);
            tids = insertAtIndex(tids, new int[]{-1, -2}, index);
        }

        if (warning != null) {
            // System.out.println(G.get("season"), warning);
            LogEvent.logEvent(new LogEventData("info", 
                String.format("Your <a href=\"%s\">schedule settings (# Games, # Division Games, and # Conference Games)</a> combined with your teams/divs/confs cannot be handled by the schedule generator, so instead it will generate round robin matchups between all your teams. Message from the schedule generator: \"%s\" <a href=\"https://%s/manual/customization/schedule-settings/\" target=\"_blank\">More details.</a>",
                Helpers.leagueUrl(new String[]{"settings"}), warning, WEBSITE_ROOT),
                false), conditions);
        }

        return tids;
    }

    private static int[][] insertAtIndex(int[][] array, int[] element, int index) {
        int[][] newArray = new int[array.length + 1][];
        System.arraycopy(array, 0, newArray, 0, index);
        newArray[index] = element;
        System.arraycopy(array, index, newArray, index + 1, array.length - index);
        return newArray;
    }

    static class Team {
        SeasonAttributes seasonAttrs;
        int tid;

        static class SeasonAttributes {
            int cid;
            int did;
        }
    }
}

File: src/worker/core/season/newSchedule.clj
(ns my-namespace
  (:require [common.index :refer [WEBSITE_ROOT]]
            [common.types :as types]
            [util.index :refer [g helpers log-event]]
            [newScheduleGood :refer [new-schedule-good]]))

(defn new-schedule [teams conditions]
  (let [{:keys [tids warning]} (new-schedule-good teams)]

    ;; Add trade deadline
    (let [trade-deadline (g/get "tradeDeadline")]
      (when (< trade-deadline 1)
        (let [index (Math/round (* (helpers/bound trade-deadline 0 1) (count tids)))]
          (assoc tids index [-3 -3]))))

    ;; Add an All-Star Game
    (let [all-star-game (g/get "allStarGame")]
      (when (and (not (nil? all-star-game)) (>= all-star-game 0))
        (let [index (Math/round (* (helpers/bound all-star-game 0 1) (count tids)))]
          (assoc tids index [-1 -2]))))

    (when (not (nil? warning))
      ;; console.log(g.get("season"), warning);
      (log-event
        {:type "info"
         :text (str "Your <a href=\"" (helpers/league-url ["settings"]) "\">schedule settings (# Games, # Division Games, and # Conference Games)</a> combined with your teams/divs/confs cannot be handled by the schedule generator, so instead it will generate round robin matchups between all your teams. Message from the schedule generator: \"" warning "\" <a href=\"https://" WEBSITE_ROOT "/manual/customization/schedule-settings/\" target=\"_blank\">More details.</a>")
         :save-to-db false}
        conditions))

    tids))
